src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -335,5 +333,114 @@ private static string OrderLocationAttributes(LocationAttributeNode node)
+                    requestFilterGroups,
+                    filterGroup => filterGroup.Id,
+                    requestFilterGroup => requestFilterGroup.Id,
+                    $"Requested filter groups do not match subject filter groups for filter: {filter.Id}")
+                .OnSuccess(_ =>
+                {
+                    var requestMap = requestFilterGroups.ToDictionary(filterGroup => filterGroup.Id);
+                    var results = filter.FilterGroups.Select(filterGroup =>
+                            AssertCollectionsAreSameIgnoringOrder(filterGroup.FilterItems,
+                                requestMap[filterGroup.Id].FilterItems,
+                                indicator => indicator.Id,
+                                $"Requested filter items do not match subject for filter group: {filterGroup.Id}"))
+                        .ToList();
+
+                    var failure = results.FirstOrDefault(result => result.IsLeft)?.Left;
+                    return failure ?? new Either<ActionResult, Unit>(Unit.Instance);
+                });
+        }
+
+        private async Task<Either<ActionResult, Unit>> ValidateIndicatorGroupsForSubject(
+            Guid subjectId,
+            List<IndicatorGroupUpdateViewModel> requestIndicatorGroups)
+        {
+            var indicatorGroups = await _indicatorGroupRepository.GetIndicatorGroups(subjectId);
+            return AssertCollectionsAreSameIgnoringOrder(indicatorGroups,
+                    requestIndicatorGroups,
+                    indicatorGroup => indicatorGroup.Id,
+                    requestIndicatorGroup => requestIndicatorGroup.Id,
+                    "Requested indicator groups do not match subject indicator groups")
+                .OnSuccess(_ =>
+                {
+                    var requestMap = requestIndicatorGroups.ToDictionary(indicatorGroup => indicatorGroup.Id);
+                    var results = indicatorGroups.Select(indicatorGroup =>
+                            AssertCollectionsAreSameIgnoringOrder(indicatorGroup.Indicators,
+                                requestMap[indicatorGroup.Id].Indicators,
+                                indicator => indicator.Id,
+                                $"Requested indicators do not match subject for indicator group: {indicatorGroup.Id}"))
+                        .ToList();
+
+                    var failure = results.FirstOrDefault(result => result.IsLeft)?.Left;
+                    return failure ?? new Either<ActionResult, Unit>(Unit.Instance);
+                });
+        }
+
+        private static Either<ActionResult, Unit> AssertCollectionsAreSameIgnoringOrder<TLeft, TRight, TId>(
+            IEnumerable<TLeft> left,
+            IEnumerable<TRight> right,
+            Func<TLeft, TId> leftIdSelector,
+            Func<TRight, TId> rightIdSelector,
+            string error)
+        {
+            var rightIdList = right.Select(rightIdSelector).ToList();
```

Comment:
I tend to equate "left" and "right" with error handling.  Mayeb change the terminology slightly?  First, Second?  TFirst, TSecond?

Reply:
Renamed to first/second/TFirst/TSecond etc, thanks.

Reply:
Cool that works better for me personally, ta!